vantage6/vantage6/cli/utils_kubernetes.py

```diff
@@ -0,0 +1,247 @@
+"""
+Kubernetes utility functions for Vantage6 CLI.
+
+This module provides utilities for handling Kubernetes client configuration,
+especially for MicroK8s environments that may have SSL certificate issues.
+"""
+
+import base64
+import ssl
+from pathlib import Path
+from typing import Optional
+
+from kubernetes import client, config
+from kubernetes.config.config_exception import ConfigException
+
+from vantage6.common import warning
+
+
+def configure_kubernetes_client_for_microk8s(cfg: client.Configuration) -> None:
+    """
+    Configure the Kubernetes client to handle MicroK8s SSL certificate issues.
+
+    This function detects if we're using MicroK8s and configures the client
+    to handle SSL certificates appropriately for both development and production.
+
+    Parameters
+    ----------
+    cfg : client.Configuration
+        The Kubernetes client configuration to configure
+    """
+    # Check if we're using MicroK8s by looking at the current context
+    if is_microk8s_context():
+        _configure_microk8s_ssl(cfg)
+
+
+def _configure_microk8s_ssl(cfg: client.Configuration) -> None:
+    """
+    Configure SSL settings for MicroK8s.
+
+    This function handles MicroK8s SSL certificates in a secure way by:
+    1. First trying to use the MicroK8s certificate directly
+    2. If that fails, falling back to a more lenient but still secure approach
+    """
+    try:
+        # Try to get the MicroK8s certificate and use it properly
+        cert_path = _get_microk8s_certificate_path()
+        if cert_path and cert_path.exists():
```

Comment:
The `_get_microk8s_certificate_path` already garantees that the path exists. Can't really harm but there is also no reason.
```suggestion
        cert_path = _get_microk8s_certificate_path()
        if cert_path:
```